rre-dataset-generator/src/search_engine/solr_search_engine.py

```
@@ -0,0 +1,85 @@
+                json_body[key] = [default_value]
+
+        return {
+            'query': json_body.get('q')[0],
+            'params': {k: v[0] for k, v in json_body.items() if k != 'q'}
+        }
+
+    def extract_documents_to_generate_queries(self,
+                                             documents_filter: Union[None, List[Dict[str, List[str]]]],
+                                             doc_number: int) \
+            -> List[Dict[str, Any]]:
+        payload = {
+            'query': '*:*',
+            'params': {
+                'rows': doc_number
+            }
+        }
+
+        if documents_filter is not None:
+            payload['params']['fq'] = []
+            for dict_field in documents_filter:
+                for field, values in dict_field.items():
+                    if not values:
+                        continue  # skip empty lists
+                    if len(values) == 1:
+                        clause = f'{field}:{values[0]}'
+                    else:
+                        or_values = ' OR '.join(f'{v}' for v in values)
+                        clause = f'{field}:({or_values})'
+                    payload['params']['fq'].append(clause)
+
+        return self.search(payload)
+
+    def extract_documents_to_evaluate_system(self, query_template: str, keyword: str="*:*") -> List[Dict[str, Any]]:
+        """Search for documents using a query."""
+        template = query_template.replace(self.PLACEHOLDER, keyword)
+        payload = self.template_to_json_body(template)
+        return self.search(payload)
+
+    def search(self, payload: Dict[str, Any]) -> List[Dict[str, Any]]:
+        """Search for documents using a query."""
+        search_url = os.path.join(str(self.endpoint), "select")
+
+        response = requests.post(search_url, headers=self.HEADERS, json=payload)
+        if response.status_code == 200:
+            return response.json()['response']['docs']
+        else:
+            raise ValueError()
```

ValueError is very vague, better to throw an error with http code, if it is server error or client and also log error?

---

I'll add more error handling (to review) in the next commit

---

I'd better add log.error and keeping ValueError() or some try-except, than defining custom Exceptions.